### PR TITLE
Feature/propwidget

### PR DESCRIPTION
**Pull request by petersteneteg**
Disable and enable layout in the propertylists when doing many changes. 10X perf in some cases.
PropertyOwner got a clear() and empty() method.

### CODE REVIEW

**Review by ResearchDaniel — DISMISSED** [review]
What’s the reason for changing the depth format to 32-bit?

**Comment by petersteneteg** [comment]
We use a float32 on the cpu side, so to match that one. And I ran into some precision issues..

File: modules/python3/bindings/include/inviwopy/vectoridentifierwrapper.h [hunk 1]
```diff
@@ -59,11 +59,21 @@ class VectorIdentifierWrapper {
...
+        if (pos >= 0) {
+            const auto ind = static_cast<size_t>(pos);
+            if (ind < vector_.size()) {
+                return *vector_[ind];
+            } else {
+                throw pybind11::index_error();
+            }
         } else {
-            throw pybind11::index_error();
+            const auto ind = static_cast<ptrdiff_t>(vector_.size()) + pos;
```
Comment (martinfalk) [inline modules/python3/bindings/include/inviwopy/vectoridentifierwrapper.h:71]: that's neat

File: modules/qtwidgets/src/propertylistwidget.cpp [hunk 2]
```diff
@@ -195,6 +205,16 @@ bool PropertyListWidget::event(QEvent* e) {
     }
 }
 
+void PropertyListWidget::onProcessorNetworkEvaluationBegin() { 
+    setUpdatesEnabled(false); 
+    listLayout_->setEnabled(false);
+}
```
Comment (martinfalk) [inline modules/qtwidgets/src/propertylistwidget.cpp:211]: this should take care of most of the issues we encountered so for, shouldn't it? Nice.

**Review by martinfalk — DISMISSED** [review]

**Review by martinfalk — APPROVED** [review]

**Comment by petersteneteg** [comment]
@martinfalk I think this merge resulted in a quite substantial perf regression, My guess is that it is related to the move to 32 bit depth, does that make sense to you?

**Comment by martinfalk** [comment]
Not sure. That would need performance testing in particular with different hardware. In some older posts I found that AMD GPUs did not support 24bits at all...

See also https://www.khronos.org/opengl/wiki/Common_Mistakes#Depth_Buffer_Precision

**Comment by petersteneteg** [comment]
this for the ray caster with animation regression test:
image
the drop is when the bitdepth changed. see the last set of runs on http://jenkins.inviwo.org:8080/job/tests/job/hist/Regression/

But maybe we should consider other hardware too?

**Pull request merged**